repos/system_upgrade/el7toel8/actors/peseventsscanner/libraries/peseventsscanner.py

```diff
@@ -475,44 +478,55 @@ def map_repositories(packages):
+                                "the Red Hat Enterprise Linux 7 product.")
 
 
-def report_skipped_packages(message, packages):
+def report_skipped_packages(message, packages, remediation=None):
     """Generate report message about skipped packages"""
     packages = sorted(packages)
     title = 'Packages will not be installed'
     summary = '{} {}\n{}'.format(len(packages), message, '\n'.join(['- ' + p for p in packages]))
-    reporting.create_report([
+    report_content = [
         reporting.Title(title),
         reporting.Summary(summary),
         reporting.Severity(reporting.Severity.HIGH),
         reporting.Tags([reporting.Tags.REPOSITORY]),
-    ] + [reporting.RelatedResource('package', p) for p in packages])
+    ]
+    if remediation:
+        report_content += [reporting.Remediation(hint=remediation)]
+    report_content += [reporting.RelatedResource('package', p) for p in packages]
+    reporting.create_report(report_content)
     if is_verbose():
-        api.show_message(summary)
+        api.current_logger().info(summary)
 
 
 def add_output_pkgs_to_transaction_conf(transaction_configuration, events):
     """
-    Add more packages for removal to transaction configuration if they can be derived as outputs of PES events.
+    Filter out those PES events that conflict with the higher priority transaction configuration files.
 
     Output packages from an event are added to packages for removal only if all input packages are already there.
 
     :param transaction_configuration: RpmTransactionTasks model instance with pkgs to install, keep and remove based
                                       on the user configuration files
     :param events: List of Event tuples, where each event contains event type and input/output pkgs
     """
-    message = 'Marking packages for removal:\n'
+    message = 'The following RHEL 8 packages will not be installed:\n'
 
     for event in events:
         if event.action in (Action.split, Action.merged, Action.replaced, Action.renamed):
             if all([pkg in transaction_configuration.to_remove for pkg in event.in_pkgs]):
                 transaction_configuration.to_remove.extend(event.out_pkgs)
-                message += '- [{action}] {ins} -> {outs}\n'.format(
-                    action=event.action.name,
-                    ins=', '.join(sorted(event.in_pkgs.keys())),
-                    outs=', '.join(sorted(event.out_pkgs.keys()))
+                message += (
+                    '- {outs}\n - Reason: {ins} being {action} {to_or_by} {outs} {is_or_are} mentioned in the'
+                    ' transaction configuration file /etc/leapp/transaction/to_remove\n'.format(
+                        outs=', '.join(event.out_pkgs.keys()),
+                        ins=', '.join(event.in_pkgs.keys()),
+                        action=event.action.name,
+                        to_or_by='by' if event.action == 'Replaced' else 'by',
```

Comment:
Isn't this always `'by'`?

Reply:
Thx for catching. However I think it should be by only in case of `Replaced by`. Merged to, renamed to, split to.